modules/codegen/src/smithy4s/codegen/SmithyToIR.scala

```diff
@@ -480,9 +525,21 @@ private[codegen] class SmithyToIR(model: Model, namespace: String) {
   private object UnRef {
     def unapply(tpe: Type): Option[Shape] = tpe match {
       case Type.Ref(ns, name) =>
-        model
+        val maybeShape = model
           .getShape(ShapeId.fromParts(ns, name))
           .asScala
+        maybeShape.map { shape =>
+          shape.getTrait(classOf[AdtMemberTrait]).asScala match {
+            case Some(adtMemberTrait) =>
+              val cId = shape.getId
+              val newNs =
+                cId.getNamespace + "." + adtMemberTrait.getValue.getName
+              shape.asStructureShape.get.toBuilder
+                .id(ShapeId.fromParts(newNs, cId.getName))
+                .build()
+            case _ => shape
+          }
+        }
```

Comment:
Alright.. so this is kind of hackish, so let me know if anyone can think of a better way. Basically, if the shape contains the adtmember trait, then I am rewriting its ShapeId so that it will be moved into the correct location (add the name of the union to the namespace). It works, but 😬